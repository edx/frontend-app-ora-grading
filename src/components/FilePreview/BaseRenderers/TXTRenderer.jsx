import React, { useMemo, useState } from 'react';
import PropTypes from 'prop-types';
import { get } from 'axios';

const TXTRenderer = ({ url, onError, onSuccess }) => {
  const [content, setContent] = useState('');
  useMemo(() => {
    get(url)
      .then(({ data }) => {
        onSuccess();
        setContent(data);
      })
      .catch(({ response }) => onError(response.status));
  }, [url]);

  return (
    <pre className="txt-renderer">
      {content}
    </pre>
  );
};

TXTRenderer.defaultProps = {};

TXTRenderer.propTypes = {
  url: PropTypes.string.isRequired,
  onError: PropTypes.func.isRequired,
  onSuccess: PropTypes.func.isRequired,
};

export default TXTRenderer;
