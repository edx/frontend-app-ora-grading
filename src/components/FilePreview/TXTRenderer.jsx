import React, { useMemo, useState } from 'react';
import PropTypes from 'prop-types';
import { get } from 'data/services/lms/utils';

const TXTRenderer = ({ url }) => {
  const [content, setContent] = useState('');
  useMemo(() => {
    get(url).then(({ data }) => setContent(data));
  }, [url]);

  return (
    <pre className="txt-renderer">
      {content}
    </pre>
  );
};

TXTRenderer.defaultProps = {};

TXTRenderer.propTypes = {
  url: PropTypes.string.isRequired,
};

export default TXTRenderer;
