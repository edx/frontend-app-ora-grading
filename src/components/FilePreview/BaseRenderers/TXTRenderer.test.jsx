import React from 'react';
import { shallow } from 'enzyme';

import TXTRenderer from './TXTRenderer';

jest.mock('axios', () => ({
  get: jest.fn((...args) => Promise.resolve({ data: `Content of ${args}` })),
}));

describe('TXY Renderer Component', () => {
  const props = {
    url: 'some_url.txt',
  };

  props.onError = jest.fn().mockName('onError');
  props.onSuccess = jest.fn().mockName('onSuccess');

  let el;
  beforeEach(() => {
    el = shallow(<TXTRenderer {...props} />);
  });
  test('snapshot', () => {
    expect(el).toMatchSnapshot();
  });
});
