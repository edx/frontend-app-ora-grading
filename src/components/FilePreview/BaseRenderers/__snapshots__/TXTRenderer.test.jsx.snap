// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TXY Renderer Component snapshot 1`] = `
<pre
  className="txt-renderer"
>
  Content of some_url.txt
</pre>
`;
